controllers/liveupdate_controller.go

```diff
@@ -0,0 +1,653 @@
+	if err != nil || len(podList.Items) == 0 {
+		reqLogger.Info("No pods found")
+		return ctrl.Result{}, nil
+	}
+
+	outdated, err := isImageOutdated(podList)
+
+	if err != nil {
+		reqLogger.Error(err, "failed to check image in the operator")
+		return ctrl.Result{}, err
+	}
+
+	if !outdated {
+		return ctrl.Result{}, err
+	}
+
+	return liveUpdateRequest.handleMigration()
+}
+
+func (r *LiveUpdateRequest) handleMigration() (ctrl.Result, error) {
+	ispn := r.infinispan
+	logger := r.reqLogger
+	switch ispn.Status.LiveUpgradeStage {
+	case ispnv1.LiveUpgradeStart:
+		logger.Info("Start LiveUpgrade")
+		return r.prepare()
+	case ispnv1.LiveUpgradePrepared:
+		logger.Info("LiveUpgrade prepared")
+		return r.redirectService()
+	case ispnv1.LiveUpgradeRouteRedirected:
+		logger.Info("LiveUpgrade redirected")
+		return r.syncData()
+	case ispnv1.LiveUpgradeSynced:
+		logger.Info("LiveUpgrade synced")
+		return r.destroy()
+	default:
+		logger.Info("LiveUpgrade Create cluster")
+		return r.createNewStatefulSet()
+	}
+}
+
+// createNewStatefulSet Creates a new statefulSet to migrate the existing cluster to
+func (r *LiveUpdateRequest) createNewStatefulSet() (ctrl.Result, error) {
+	// Reconcile configmap with new DNS query
+	if err := r.reconcileNewConfigMap(); err != nil {
+		return reconcile.Result{}, err
+	}
+	// Reconcile ping service with new DNS query
+	if err := r.reconcileNewPingService(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Obtain existing statefulSet
+	namespace := r.infinispan.Namespace
+	statefulSetName := r.infinispan.GetStatefulSetName()
+	sourceStatefulSet := &appsv1.StatefulSet{}
+	err := r.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: statefulSetName}, sourceStatefulSet)
+
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "Could not find existing statefulSet'%s'", statefulSetName)
+		return reconcile.Result{}, err
+	}
+
+	// Create a new statefulSet based on the existing one
+	targetStatefulSet := &appsv1.StatefulSet{}
+	err = r.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: r.targetName}, targetStatefulSet)
+	if err != nil && errors.IsNotFound(err) {
+		targetStatefulset := sourceStatefulSet.DeepCopy()
+		volumes := targetStatefulset.Spec.Template.Spec.Volumes
+		// Change the config name
+		for _, volume := range volumes {
+			if volume.Name == ConfigVolumeName {
+				volume.ConfigMap.Name = r.newConfigName
+			}
+		}
+		targetStatefulset.Name = r.targetName
+		targetStatefulset.ObjectMeta.ResourceVersion = ""
+		// Update the image references
+		container := &targetStatefulset.Spec.Template.Spec.Containers[0]
+		container.Image = consts.DefaultImageName
+		env := container.Env
+		for o, envVar := range env {
+			if envVar.Name == "DEFAULT_IMAGE" {
+				a := corev1.EnvVar{
+					Name:      envVar.Name,
+					Value:     consts.DefaultImageName,
+					ValueFrom: envVar.ValueFrom,
+				}
+				env[o] = a
+				break
+			}
+		}
+		container.Env = env
```

Comment:
If we use `container.Env` directly then there's no need for this reassignment or the `env` var.